internal/datacoord/channel_manager.go

```diff
@@ -778,27 +778,32 @@ func (c *ChannelManager) Release(nodeID UniqueID, channelName string) error {
 
 // Reassign reassigns a channel to another DataNode.
 func (c *ChannelManager) Reassign(originNodeID UniqueID, channelName string) error {
-	c.mu.Lock()
-	defer c.mu.Unlock()
-
+	c.mu.RLock()
 	ch := c.getChannelByNodeAndName(originNodeID, channelName)
 	if ch == nil {
+		c.mu.RUnlock()
 		return fmt.Errorf("fail to find matching nodeID: %d with channelName: %s", originNodeID, channelName)
 	}
+	c.mu.RUnlock()
```

Comment:
this seems to be not a right solution

if we unlock the channel this might be removed during this period?

Reply:
it will get channel information again before generating the reassign policy.  it doesn't care about whether the lock channel for channel is dropped status.